### PR TITLE
Decaying turbulence

### PR DESCRIPTION
**Pull request by McBs**
Hello,
I want to add a new flow, which is defined as "decaying turbulence". It is implemented in 2D and 3D.
The tests verify that the divergence is zero and that the initial energy given is maintained after initialization
The initialization was done with the help of Holger Foysi.

### CODE REVIEW

**Comment by Olllom** [comment]
Looks good to me at first glance. I'll wait for the CI to finish. 
@dominikwilde could you review this?

File: lettuce/flows/__init__.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 from lettuce.flows.couette import CouetteFlow2D
 from lettuce.flows.poiseuille import PoiseuilleFlow2D
 from lettuce.flows.doublyshear import DoublyPeriodicShear2D
+from lettuce.flows.decayingturbulence import DecayingTurbulence2D, DecayingTurbulence3D
```
Comment (Olllom) [inline lettuce/flows/__init__.py:9]: Maybe add them as "decaying2D" and "decaying3D" to the `flow_by_name` dictionary

**Comment by dominikwilde** [comment]
Alright, this look good to me - although I am not the expert here.

However, I have some minor comments:

1. The pressure is uniformly set to 0 ... Is that appropriate after all the effort to cancel the divergence?
2. You should add even more comments - otherwise it will be hard for someone else to follow the steps (and eventually for yourself). For example, explain the meaning of k0.
3. Give a source from which the spectrum function is taken (lines 37 and 152)
4. Stay consistent with the variable names (for example you mix _r and _real for the real parts of the velocity, the same applies for Fourier-transformed variables).

`        ucr = uc.real`
`u0_real_h = ...` 

Very nice contribution!

**Comment by Olllom** [comment]
Thanks. All good remarks.
Pressure initialization is important

**Comment by McBs** [comment]
Thanks for the comments.

I implemented a Jacobi scheme for the calculation of the pressure-Poisson equation. This should lead to less shock waves in the beginning! In addition, the syntax has been improved. Please have a look at this.

**Comment by dominikwilde** [comment]
Looks good.
There are some unneeded comments in the util.py left that you should remove before the merge. The rest is fine.

` #dx = self.units.convert_length_to_pu(1)`
`# f = torch.tensor(f, device=device, dtype=torch.double)`

**Comment by McBs** [comment]
Thanks for your commend. Syntax has been improved, the conflict has been solved and I added some comments.

File: lettuce/flows/decayingturbulence.py [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+        u1 = u1_real + u1_imag * 1.0j
+        u1f = np.fft.ifftn(u1, axes=(0, 1, 2)) * norm
+        u = np.append(u, u1f.real[None, ...], axis=0)
+
+        # backtransformation of u2
+        u2 = u2_real + u2_imag * 1.0j
+        u2f = np.fft.ifftn(u2, axes=(0, 1, 2)) * norm
+        u = np.append(u, u2f.real[None, ...], axis=0)
+
+        p = (u[0]*0)[None,...]
```
Comment (Olllom) [inline lettuce/flows/decayingturbulence.py:273]: why is the pressure not initialized with self.pressure_poisson?

**Pull request merged**